src/cljam/io/wig.clj

```diff
@@ -0,0 +1,171 @@
+
+(defn ^WIGWriter writer
+  "Returns an open cljam.io.wig.WIGWriter of f. Should be used inside with-open
+  to ensure the writer is properly closed."
+  [f]
+  (WIGWriter. (cio/writer (util/compressor-output-stream f)) (util/as-url f)))
+
+(defn- header-or-comment?
+  "Checks if given string is neither a header nor a comment line."
+  [^String s]
+  (or (empty? s)
+      (.startsWith s "browser")
+      (.startsWith s "track")
+      (.startsWith s "#")))
+
+(defn- normalize
+  "Normalize WIG lines."
+  [m]
+  (update m :chrom chr/normalize-chromosome-key))
+
+(defn- fields->map
+  "Convert vector [\"key1=value1\" \"key2=value2\" ...] to map {:key1 \"value1\" :key2 \"value2\" ...}."
+  [fields]
+  (->> fields
+       (map #(cstr/split % #"="))
+       (map (fn [[k v]] [(keyword k) v]))
+       (into {})))
+
+(defn- str->wiggle-track-data
+  "Convert string to Wiggle track data values which can be integer or real,
+  positive or negative values."
+  [^String s]
+  (if-let [l (as-long s)]
+    l
+    (as-double s)))
+
+(defn- deserialize-wigs
+  "Parse WIG lines and returns a vector of maps."
+  [lines]
+  (loop [lines lines
+         wigs []
+         current-track nil]
+    (if (nil? (first lines))
+      (conj wigs current-track)
+      (let [fields (-> lines first cstr/trim (cstr/split #"\s+"))]
+        (case (first fields)
+          ; declaration line of variableStep
+          "variableStep"
+          (let [{:keys [chrom span] :or {span 1}} (->> fields rest fields->map)]
+            (recur (rest lines)
+                   (cond-> wigs
+                     (some? current-track) (conj current-track))
+                   {:format :variable-step
+                    :chrom chrom
+                    :span (as-long span)
+                    :values []}))
+
+          ; declaration line of fixedStep
+          "fixedStep"
+          (let [{:keys [chrom start step span] :or {span 1}} (->> fields rest fields->map)]
+            (recur (rest lines)
+                   (cond-> wigs
+                     (some? current-track) (conj current-track))
+                   {:format :fixed-step
+                    :chrom chrom
+                    :start (as-long start)
+                    :step (as-long step)
+                    :span (as-long span)
+                    :values []}))
+
+          ; data line
+          (let [v (case (:format current-track)
+                    :variable-step
+                    (let [[start value] fields]
+                      {:start (as-long start)
+                       :value (str->wiggle-track-data value)})
+
+                    :fixed-step
+                    (-> fields first str->wiggle-track-data)
+
+                    (throw (IllegalArgumentException. "Invalid wiggle format")))]
+            (recur (rest lines)
+                   wigs
+                   (update current-track :values #(conj % v)))))))))
+
+(defn- serialize-wig
+  "Serialize WIG line into vector of strings."
```

Comment:
The docstring doesn’t seem to describe what the function does, in the sense that it doesn’t return “a vector of strings”.